syntax = "proto3";

package com.webank.ai.fate.api.mlmodel.manager;
option java_outer_classname = "ModelServiceProto";

message ModelInfo{
    string name = 1;
    string namespace = 2;
}

message PublishRequest{
    string sceneId = 1;
    string myRole = 2;
    string myPartyId = 3;
    string partnerPartyId = 4;
    string modelName = 5;
    string modelNamespace = 6;
}

message PublishResponse{
    int32 statusCode = 1;
    string message = 2;
    string error = 3;
    bytes data = 4;
}


service ModelService{
    rpc publishLoad(PublishRequest) returns (PublishResponse);
    rpc publishOnline(PublishRequest) returns (PublishResponse);
}
